var/spack/repos/builtin/packages/compiler-wrapper/package.py

```diff
@@ -0,0 +1,259 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import pathlib
+import shutil
+import sys
+from typing import List
+
+import archspec.cpu
+
+from llnl.util import lang
+
+import spack.compilers.libraries
+import spack.package_base
+from spack.package import *
+
+
+class CompilerWrapper(Package):
+    """Spack compiler wrapper script.
+
+    Compiler commands go through this compiler wrapper in Spack builds.
+    The compiler wrapper is a thin layer around the standard compilers.
+    It enables several key pieces of functionality:
+
+    1. It allows Spack to swap compilers into and out of builds easily.
+    2. It adds several options to the compile line so that spack
+       packages can find their dependencies at build time and run time:
+       -I and/or -isystem arguments for dependency /include directories.
+       -L                 arguments for dependency /lib directories.
+       -Wl,-rpath         arguments for dependency /lib directories.
+    """
+
+    homepage = "https://github.com/spack/spack"
+    url = f"file:///{pathlib.PurePath(__file__).parent}/cc.sh"
+
+    # FIXME (compiler as nodes): use a different tag, since this is only to exclude
+    # this node from auto-generated rules
+    tags = ["runtime"]
+
+    license("Apache-2.0 OR MIT")
+
+    if sys.platform != "win32":
+        version(
+            "1.0",
+            sha256="84a26f8f37329bcdfb41d23a8a4f4c46efbede983eb73737135c95a4c126b5b7",
```

Comment:
How about:

```suggestion
    path = f"{pathlib.PurePath(__file__).parent}/cc.sh"
    url = f"file:///{path}"

    # FIXME (compiler as nodes): use a different tag, since this is only to exclude
    # this node from auto-generated rules
    tags = ["runtime"]

    license("Apache-2.0 OR MIT")

    if sys.platform != "win32":
        version(
            "1.0",
            sha256=hashlib.sha256(open(path, "rb").read()).hexdigest,
```

Reply:
Do we want to open a file and compute a sha256 each time we run Spack?

Reply:
I'm not really a fan of opening a file and computing a sha256 each time we define this class. It's probably better to update manually for now.

Reply:
It's very easy to compute a sha256.  But sure, better to update it at fetch time like we do for repo-resident patches.  Maybe something to leave for immediately after this PR.  I don't like the maintenance burden of manually computing a hash for a file in the repo but ok let's leave it for now.